tests/general/internal_metrics_test.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+
+package tests
+
+import (
+	"testing"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+func TestDefaultInternalMetrics(t *testing.T) {
+	testutils.RunMetricsCollectionTest(t, "internal_metrics_config.yaml", "internal_metrics_expected.yaml",
+		testutils.WithCompareMetricsOptions(
+			pmetrictest.IgnoreStartTimestamp(),
+			pmetrictest.IgnoreTimestamp(),
+			pmetrictest.IgnoreMetricsOrder(),
+			pmetrictest.IgnoreScopeVersion(),
+			pmetrictest.IgnoreMetricValues(
+				"otelcol_exporter_sent_metric_points",
+				"otelcol_process_cpu_seconds",
+				"otelcol_process_memory_rss",
+				"otelcol_process_runtime_heap_alloc_bytes",
+				"otelcol_process_runtime_total_alloc_bytes",
+				"otelcol_process_runtime_total_sys_memory_bytes",
+				"otelcol_process_uptime",
+				"otelcol_receiver_accepted_metric_points",
+				"scrape_duration_seconds",
+				"scrape_samples_post_metric_relabeling",
+				"scrape_samples_scraped",
+				"scrape_series_added",
+				"up",
+			),
+			pmetrictest.IgnoreResourceAttributeValue("service_instance_id"),
+			pmetrictest.IgnoreResourceAttributeValue("service_version"),
+			pmetrictest.IgnoreMetricAttributeValue("service_instance_id"),
+			pmetrictest.IgnoreMetricAttributeValue("service_version"),
```

Comment:
Entirely unrelated to this PR, but do you know why these attributes are on both the resource and data point?

Reply:
It's a migration phase. We used to be able to set it on data point attributes (open sensus instrumentation). Now, it's partially migrated to the otel instrumentation where we can set the resource attributes. The next release will be only setting resource attributes.

Reply:
Thanks, makes sense